File: examples/cxx/test_functions/models/levi13.cpp
#include "levi13.h"
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

// Minus Levi13
void m_levi13(std::vector<double>& pars, std::vector<double>& result)
{
 size_t nPars = pars.size();
 if( nPars != 2 ) { printf("levi13: error, num pars must be 2 (is %zu)\n", nPars); exit(-1); }

 double s = pow(sin(3*M_PI*pars[0]),2)+pow((pars[0]-1),2)*(1.0+pow(sin(3*M_PI*pars[1]),2))+pow(pars[1]-1,2)*(1.0+pow(sin(2.0*M_PI*pars[1]),2));

 result.push_back(-s);
}

File: examples/python/test_functions/model/levi13.py
#!/usr/bin/env python
import math
    
# Minus Levi13
def m_levi13(x):
    par1 = x.getParameter(0)
    par2 = x.getParameter(1)

    result = pow(math.sin(3.0*math.pi*par1),2)+pow((par1-1.0),2)*(1.0+pow(math.sin(3*math.pi*par2),2))+pow(par2-1,2)*(1.0+pow(math.sin(2.0*math.pi*par2),2))
    x.addResult(-result)
